packages/integration/src/gerrit/core.ts

```diff
@@ -156,14 +157,23 @@ export function getAuthenticationPrefix(
 export function getGitilesAuthenticationUrl(
   config: GerritIntegrationConfig,
 ): string {
-  const parsedUrl = new URL(config.gitilesBaseUrl!);
-  parsedUrl.pathname = parsedUrl.pathname.replace(/\/?$/, '');
-  parsedUrl.pathname = parsedUrl.pathname.replace(
-    /\/([^\/]*)$/,
-    `${getAuthenticationPrefix(config)}$1`,
-  );
-
-  return parsedUrl.toString();
+  if (!config.baseUrl || !config.gitilesBaseUrl) {
+    throw new Error(
+      'Unexpected Gerrit config values. baseUrl or gitilesBaseUrl not set.',
+    );
+  }
+  if (config.gitilesBaseUrl.startsWith(config.baseUrl)) {
+    return config.gitilesBaseUrl.replace(
+      config.baseUrl.concat('/'),
+      config.baseUrl.concat(getAuthenticationPrefix(config)),
+    );
+  }
+  if (config.password) {
+    logger.warning(
```

Comment:
Can we throw an error for this case instead, or is it fine to ignore? We really do not want to be logging here tbh